src/packages/helpers/src/resolver.ts

```diff
@@ -0,0 +1,76 @@
+import { GraphQLEntity, createBaseResolver, BackendProvider } from '@exogee/graphweaver';
+import { Resolver as ResolverDecorator, Field, ObjectType, ID } from 'type-graphql';
+
+import { caps, createFieldOnClass, setNameOnClass, setClassReadOnly } from './util';
+
+interface Item {
+	id: string;
+	[key: string]: string | number | boolean;
+}
+
+export interface FieldOptions<TEntity> {
+	name: string;
+	type: 'string' | 'float' | 'boolean';
+	resolve?(data: TEntity, fieldName: string): any;
+	optional?: boolean;
+	metadata?: Record<string, any>;
+}
+
+export interface ResolverOptions<TEntity extends Item> {
+	name: string;
+	fields: Array<FieldOptions<TEntity>>;
+	provider: BackendProvider<TEntity, TEntity>;
+	readOnly?: boolean;
+}
+
+export const createResolver = <TEntity extends Item>({
+	name,
+	fields,
+	provider,
+	readOnly,
+}: ResolverOptions<TEntity>) => {
+	const entityName = caps(name);
+
+	// Create GraphQL Entity
+	@ObjectType(entityName)
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-expect-error @todo need to fix the type issue heree
+	class Entity extends GraphQLEntity<TEntity> {
```

Comment:
Is this because it is expecting Item to implement `BaseDataEntity` ?

Reply:
yes, it is